### PR TITLE
feat(register): new user registration

### PR DESCRIPTION
**Pull request by gilsonbridge**
updated APIs for new user registration

### CODE REVIEW

File: server/routes.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export default function routes(app: Express) {
   // Auth
   router.route('/auth/resetpwd').post(user.resetpwd);
   router.route('/auth/saml').get(auth.getActiveVendors);
+  router.route('/register').post(auth.register);
```
Comment (BobCashStory) [inline server/routes.ts:78]: Why not put it in the `auth` scope ?

File: server/controllers/AuthCtrl.ts [hunk 2]
```diff
@@ -44,13 +45,28 @@ export default class AuthCtrl extends BaseCtrl {
     }
   };
 
+  register = async (req, res: Response): Promise<any> => {
```
Comment (BobCashStory) [inline server/controllers/AuthCtrl.ts:48]: You should provide test with this one

**Review by BobCashStory — CHANGES_REQUESTED** [review]

File: server/routes.ts (hunk 1 above)
Comment (gilsonbridge) [reply server/routes.ts]: Updated in commit a909fc0

File: server/controllers/AuthCtrl.ts (hunk 2 above)
Comment (gilsonbridge) [reply server/controllers/AuthCtrl.ts:48]: Updated in commit a909fc0

**Review by BobCashStory — APPROVED** [review]

**Pull request merged**